noxfile.py

```diff
@@ -92,6 +93,9 @@ def tests(session, shard):
         env["MISTRAL_API_KEY"] = session.env.get("MISTRAL_API_KEY")
         env["OPENAI_API_KEY"] = session.env.get("OPENAI_API_KEY")
 
+    if shard == "openai_agents":
+        env["OPENAI_API_KEY"] = session.env.get("OPENAI_API_KEY", "DUMMY_API_KEY")
```

Comment:
Maybe I'm doing something wrong but this line replaces the API key to DUMMY_API_KEY even when passed explicitly like so:
`OPENAI_API_KEY=<api-key> nox -e "tests-3.13(shard='openai_agents')"`

Since this makes it impossible to re-record and I can't find a way to pass environment vars to the command directly (--env sets the virtual environment not the vars) it might be an issue.